src/hooks/requests/useTags.ts

```diff
@@ -0,0 +1,49 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query'
+import { endpoints } from 'api/endpoints'
+import { apiClient } from 'api'
+import { TagsDataType, TagsType } from 'types/tags'
+
+export const useFetchTags = () => {
+  const { isLoading, isError, data } = useQuery<TagsDataType>({
+    queryKey: ['tags'],
+    queryFn: () => apiClient.get(endpoints.TAGS),
+  })
+
+  const { data: tags } = data || {}
+
+  return {
+    isLoading,
+    isError,
+    tags,
+  }
+}
+
+export const useBulkCreate = () => {
+  const queryClient = useQueryClient()
+  const { mutateAsync: createTags, isLoading } = useMutation({
+    mutationKey: ['tags'],
+    mutationFn: async (payload: TagsType) =>
+      apiClient.post(endpoints.CREATE_TAGS, payload),
+
+    onSuccess: ({ tags }) => {
```

Comment:
**[blocker]**

Values stored incorrectly in state of react-query. This means that after adding the tags, they don't show up in the list, before we refetch all of them.

1. BE returns the new tags under "data" key, not "tags", so this "tags" will always be undefined

2. oldData here has wrong type. It is typed as "TagsType" so again with key "tags", however the actual oldData inside react-query state uses the "data" key here (Same way it is stored in useFetchTags)

3.  const newData = { ...oldData, ...tags } -- since tags is always undefined and oldData is just an object with the structure "{ data: Tags[] }" , the newData will just be exactly the same as oldData

4. As you are setting data for the "tags" key, the data needs to be in the same format as it was previously, which means the structure should be "{ data: Tags[] }". However you are returning the structure: "{ tags: { data: Tags[] } }"